### PR TITLE
Rename write_cmp_bytes to writeable_cmp_bytes

### PR DESCRIPTION
**Pull request by sffc**
See #4402, #4741, #4787

I had landed these functions but we never bikeshed their names. We _did_ bikeshed `writeable_length_hint` [here](https://docs.google.com/document/d/1zm_pQsrO7kTBjHbtmoQiriFNedUyd1_AjJ2Um3u_kD4/edit?pli=1), so I'm taking that answer and applying it to this similar function. It should be named `writeable_cmp_bytes` since it isn't actually writing anything; it's only a function on the Writeable trait.

### CODE REVIEW

**Comment by sffc** [comment]
If possible I'd like LGTMs from multiple organizations on this function name.

**Comment by sffc** [comment]
Note that we chose `writeable_length_hint` over `writeable_len_hint`, so in principle we could also choose to name this function `writeable_compare_bytes` instead of `writeable_cmp_bytes`.

File: utils/writeable/src/lib.rs [hunk 1]
```diff
@@ -351,7 +351,7 @@ macro_rules! impl_display_with_writeable {
 /// - Equality of string content
 /// - Equality of parts ([`*_parts_eq`] only)
 /// - Validity of size hint
-/// - Reflexivity of `cmp_bytes`
+/// - Reflexivity of `cmp_bytes` and order against largest and smallest strings
```
Comment (robertbastian) [inline utils/writeable/src/lib.rs:354]: nice, I was thinking about adding this

**Review by robertbastian — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

File: utils/writeable/src/lib.rs [hunk 2]
```diff
@@ -426,8 +426,14 @@ macro_rules! assert_writeable_eq {
             );
         }
         assert_eq!(actual_writeable.to_string(), $expected_str);
-        let ordering = $crate::Writeable::write_cmp_bytes(actual_writeable, $expected_str.as_bytes());
+        let ordering = $crate::Writeable::writeable_cmp_bytes(actual_writeable, $expected_str.as_bytes());
         assert_eq!(ordering, core::cmp::Ordering::Equal, $($arg)*);
+        let ordering = $crate::Writeable::writeable_cmp_bytes(actual_writeable, "\u{1FFFF}".as_bytes());
```
Comment (sffc) [inline utils/writeable/src/lib.rs:431]:
```suggestion
        let ordering = $crate::Writeable::writeable_cmp_bytes(actual_writeable, "\u{10FFFF}".as_bytes());
```

File: utils/writeable/src/try_writeable.rs [hunk 3]
```diff
@@ -386,8 +386,14 @@ macro_rules! assert_try_writeable_eq {
                 length_hint.0, actual_str.len(), format!($($arg)*),
             );
         }
-        let ordering = actual_writeable.write_cmp_bytes($expected_str.as_bytes());
+        let ordering = actual_writeable.writeable_cmp_bytes($expected_str.as_bytes());
         assert_eq!(ordering, core::cmp::Ordering::Equal, $($arg)*);
+        let ordering = actual_writeable.writeable_cmp_bytes("\u{1FFFF}".as_bytes());
```
Comment (sffc) [inline utils/writeable/src/try_writeable.rs:391]:
```suggestion
        let ordering = actual_writeable.writeable_cmp_bytes("\u{10FFFF}".as_bytes());
```

**Review by robertbastian — APPROVED** [review]

**Comment by robertbastian** [comment]
Waiting on ballot results

**Comment by sffc** [comment]
This name won the ballot. Merging.

**Pull request merged**